File: Volume 12 (1200 - 1299)/1221/1221.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    long long n, x, flag;
    cin >> n;
    while (n--) {
        flag = 0;
        cin >> x;
        if (x == 2) flag = 0;
        else if (x % 2 == 0) flag = 1;
        for (int i = 3; i < sqrt(x); i += 2)
            if (x % i == 0) flag = 1;
        flag ? cout << "Not Prime\n" : cout << "Prime\n";
    }
    return 0;
}

File: Volume 12 (1200 - 1299)/1221/1221.py
#!/usr/bin/env python3

import math

n = int(input())
for i in range(n):
	flag = 0
	x = int(input())
	if x == 2: flag = 0
	elif x % 2 == 0: flag = 1
	for i in range(3, int(math.sqrt(x)+1), 2):
		if x % i == 0:
			flag = 1
	if flag: print("Not ", end="")
	print("Prime")